stan/math/prim/mat/fun/cov_per.hpp

```diff
@@ -0,0 +1,131 @@
+#ifndef STAN_MATH_PRIM_MAT_FUN_COV_PER_HPP
+#define STAN_MATH_PRIM_MAT_FUN_COV_PER_HPP
+
+#include <stan/math/prim/scal/meta/return_type.hpp>
+#include <stan/math/prim/scal/err/check_size_match.hpp>
+#include <stan/math/prim/mat/fun/Eigen.hpp>
+#include <stan/math/prim/mat/fun/distance.hpp>
+#include <stan/math/prim/scal/err/check_not_nan.hpp>
+#include <stan/math/prim/scal/err/check_positive.hpp>
+#include <stan/math/prim/scal/fun/square.hpp>
+#include <vector>
+#include <cmath>
+
+namespace stan {
+  namespace math {
+
+    /**
+     * Returns a periodic kernel.
+     *
+     * @tparam T_x type of std::vector of elements
+     * @tparam T_sigma type of sigma
+     * @tparam T_l type of length scale
+     * @tparam T_p type of period
+     *
+     * @param x std::vector of elements that can be used in square distance.
+     *    This function assumes each element of x is the same size.
```

Comment:
It'd be worth mentioning that T_x can be a scalar, an Eigen::Vector, or an Eigen::RowVector (that's right, right?). This applies in all the other places as well I just didn't want to duplicate this comment.

Reply:
done